### PR TITLE
qxlsx/1.4.3: Add qt6 compatibility

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **qxlsx/1.4.3**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/qxlsx/all/conanfile.py [hunk 1]
```diff
@@ -52,12 +52,16 @@ def source(self):
...
     def _configure_cmake(self):
         cmake = CMake(self)
+        cmake.definitions["QT_VERSION_MAJOR"] = tools.Version(self.deps_cpp_info["qt"].version).major
         cmake.configure()
         return cmake
 
     def build(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
+        tools.replace_in_file(os.path.join(self._source_subfolder, "QXlsx", "CMakeLists.txt"),
```
Comment (prince-chrismc) [inline recipes/qxlsx/all/conanfile.py:62]: Could this be option, only if qt6 set the the required cxxstd? would be a fun surprise for qt5 consumers

Comment (MartinDelille) [reply recipes/qxlsx/all/conanfile.py:66]: Done here: 798492233.

File: recipes/qxlsx/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -8,4 +8,4 @@ find_package(QXlsx REQUIRED CONFIG)
 
 add_executable(${PROJECT_NAME} test_package.cpp)
 target_link_libraries(${PROJECT_NAME} QXlsx::Core)
-set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
+set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 17)
```
Comment (MartinDelille) [inline recipes/qxlsx/all/test_package/CMakeLists.txt:11]: @prince-chrismc How shall I handle the C++ version conditionnally here ?

File: recipes/qxlsx/all/conanfile.py (hunk 1 above)
Comment (MartinDelille) [reply recipes/qxlsx/all/conanfile.py]: Sorry the commit had to be fixed here: 329ec4c7a

**Review by SSE4 — APPROVED** [review]

**Comment by SpaceIm** [comment]
Better workaround suggested here: https://github.com/conan-io/conan-center-index/pull/8941#issuecomment-1061748791

**Comment by MartinDelille** [comment]
@SpaceIm Is the patch submitted upstream wrong ? https://github.com/QtExcel/QXlsx/pull/223

**Comment by paulharris** [comment]
I like to build with C++20, and apparently there can be some issues mixing different C++ standards between libs (I forget the details).   I think in the QT recipe, it uses validate() method to ensure the C++ is at least c++17, and then it is up the profile how far to push the standard.

Then cmake should use the C++xx standard requested by the profile/conan.

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/qxlsx/all/test_package/CMakeLists.txt (hunk 2 above)
Comment (prince-chrismc) [reply recipes/qxlsx/all/test_package/CMakeLists.txt:11]: Since this is the test packge, it's prefect

**Comment by MartinDelille** [comment]
@jgsogo Any idea why the upstream package is not built here ?

**Comment by uilianries** [comment]
@MartinDelille good news, the PR #11853 has your fix! Just need to be merged. Thanks to @ericLemanissier

**Comment by uilianries** [comment]
@MartinDelille Sorry, that PR is only related to Qt 6. You need Qt 5. So that PR does not work for you.

**Comment by MartinDelille** [comment]
@uilianries Done here: https://github.com/conan-io/conan-center-index/pull/11857

**Comment by MartinDelille** [comment]
@paulharris Is my latest patch [submitted upstream](https://github.com/QtExcel/QXlsx/pull/226) addressing the issue you are raising ?

**Comment by paulharris** [comment]
> @paulharris Is my latest patch [submitted upstream](https://github.com/QtExcel/QXlsx/pull/226) addressing the issue you are raising ?

I'll comment in there, it looks ok to me, but I wonder what the cmake experts would say about it.

**Comment by MartinDelille** [comment]
>  but I wonder what the cmake experts would say about it.

cmake expert unfortunately I'm not <(-_-)>

**Comment by MartinDelille** [comment]
I think we should wait for the next release of QXlsx because there is too much fix for Qt 6 since the last release:

[git log --oneline -- QXlsx/CMakeLists.txt]
[38c1ec8](https://github.com/QtExcel/QXlsx/commit/38c1ec8) Define cpp standard only if not defined (https://github.com/QtExcel/QXlsx/pull/226)
[e53efc5](https://github.com/QtExcel/QXlsx/commit/e53efc5) Use c++17 when using Qt 6
[73ea291](https://github.com/QtExcel/QXlsx/commit/73ea291) Minor formatting fix
[6d1667f](https://github.com/QtExcel/QXlsx/commit/6d1667f) Fix shared builds
[1b0a279](https://github.com/QtExcel/QXlsx/commit/1b0a279) CMake: Allow for OPTIONAL shared lib build (https://github.com/QtExcel/QXlsx/pull/197)
[5a086df](https://github.com/QtExcel/QXlsx/commit/5a086df) Force strict building and fix QString usage
[87afa71](https://github.com/QtExcel/QXlsx/commit/87afa71) set new cmake file for lib & console application

**Comment by MartinDelille** [comment]
On its way: https://github.com/conan-io/conan-center-index/pull/12604

**Pull request closed without merging**